maven.grammar/src/org/netbeans/modules/maven/hyperlinks/HyperlinkProviderImpl.java

```diff
@@ -453,5 +215,331 @@ private FileObject getPath(FileObject parent, String path) {
+                    Tuple tup = findProperty(text, tokenOff, documentOffset);
+
+                    if (tup != null) {
+                        String prop = tup.value.substring("${".length(), tup.value.length() - 1); //remove the brackets
+                        try {
+                            Project nbprj = getProject(doc);
+                            if (nbprj != null) {
+                                Object exRes = PluginPropertyUtils.createEvaluator(nbprj).evaluate(tup.value);
+                                if (exRes != null) {
+                                    return new String[] {prop, (String)exRes};
+                                } 
+                            } else {
+                                //pom file in repository or settings file.
+                            }
+                        } catch (ExpressionEvaluationException ex) {
+                            return null;
+                        }
+                    }
+                } else if(isMavenDependency()) {
+                    return new String[] { getMavenArtifactAbsolutePomPath() };
+                }
+            }
+            return null;
+        }
+        
+        String getMavenArtifactAbsolutePomPath() {
+            if (!isMavenDependency()) {
+                return null;
+            } else {
+                MavenEmbedder embedder = EmbedderFactory.getProjectEmbedder();
+                Artifact mavenArtifact =  embedder.createArtifact(groupId, artifactId, version, type == null ? "jar" : type);
+                return embedder.getLocalRepository().find(mavenArtifact).getFile().getAbsolutePath().replace(".jar", ".pom");
+            }
+        }
+
+        private int[] getHyperLinkSpan() {
+            if (isText) {
+                //we are in element text
+                FileObject fo = getProjectDir(doc);
+                if (fo != null && getPath(fo, ftext) != null) {
+                    return new int[] { ftokenOff, ftokenOff + ftext.length() };
+                }
+                // urls get opened..
+                if (ftext != null &&
+                        (ftext.startsWith("http://") || //NOI18N
+                        (ftext.startsWith("https://")))) { //NOI18N
+                    return new int[] { ftokenOff, ftokenOff + ftext.length() };
+                }
+                if (ftext != null) {
+                    Tuple prop = findProperty(ftext, ftokenOff, documentOffset);
+                    if (prop != null) {
+                        return new int[] { prop.spanStart, prop.spanEnd};
+                    }
+                }
+                if (isMavenDependency()) {
+                    return new int[] { ftokenOff, ftokenOff + ftext.length() };
+                }
+            }
+            return null;
+        }
+
+        private void resetSequenceToDependencyTagToken(TokenSequence<XMLTokenId> tokenSequence) {
+            while(!TokenUtilities.equals("<dependency", tokenSequence.token().text()) &&
```

Comment:
tokenSequence().token().text() materializes a String from the token, which makes usage of TokenUtilities unnecessary -- the methods in TokenUtilities work with CharSequences in order to avoid toString() conversions.